ArduCopter/wscript

```diff
@@ -34,8 +34,15 @@ def build(bld):
         use='mavlink',
     )
 
-    bld.ap_program(
-        program_name='arducopter',
-        program_groups=['bin', 'copter'],
-        use=vehicle + '_libs',
+    frames = (
+        'quad', 'tri', 'hexa', 'y6', 'octa', 'octa-quad', 'heli', 'single',
+        'coax',
     )
+    for frame in frames:
+        frame_config = frame.upper().replace('-', '_') + '_FRAME'
+        bld.ap_program(
+            program_name='arducopter-%s' % frame,
+            program_groups=['bin', 'copter'],
+            use=vehicle + '_libs',
```

Comment:
The define is only used by files under ArduCopter, so it works.

On Thu, Mar 24, 2016, 19:44 Lucas De Marchi <notifications@github.com>
wrote:

> In ArduCopter/wscript
> <https://github.com/diydrones/ardupilot/pull/3796#discussion_r57399379>:
>
> >      )
> > +    for frame in frames:
> > +        frame_config = frame.upper().replace('-', '_') + '_FRAME'
> > +        bld.ap_program(
> > +            program_name='arducopter-%s' % frame,
> > +            program_groups=['bin', 'copter'],
> > +            use=vehicle + '_libs',
>
> humn... does this work considering we would actually need a different lib
> per vehicle since changing the define below changes the compile-time
> behavior of the libraries?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/diydrones/ardupilot/pull/3796/files/04605203c60382113ab8740395a9617f99be245a..5a3239e0e6893eb6ff4aa6dc1bb5d5ad7b72e649#r57399379>
>